addon/components/o-s-s/nav-tab.ts

```diff
@@ -0,0 +1,48 @@
+import { assert } from '@ember/debug';
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+interface OSSNavTabArgs {
+  onSelection(): void;
+  tabArray: TabDefinition[];
+}
+
+interface TabDefinition {
+  icon?: string;
+  label?: string;
+  infoCircle?: boolean;
+  notificationDot?: boolean;
+  selected: boolean;
+  disabled: boolean;
+}
+
+export default class OSSNavTab extends Component<OSSNavTabArgs> {
+  constructor(owner: unknown, args: OSSNavTabArgs) {
+    super(owner, args);
+
+    assert(
+      '[component][OSS::NavTab] The parameter @onSelection of type function is mandatory',
+      typeof args.onSelection === 'function'
+    );
+
+    assert('[component][OSS::NavTab] The parameter @tabArray is mandatory', args.tabArray);
```

Comment:
maybe assert more than `typeof args.tabArray === 'array'` and that the length is `> 0` ?